File: STACK/Stack.cpp
#include<iostream>
using namespace std;
struct Node
{
    int data;
    struct Node * next;
};
typedef struct Node NODE;
typedef struct Node * PNODE;

class stackx
{
    private:
    PNODE first;
    int iCount;
    public:
    stackx()
    {
        first = NULL;
        iCount = 0;
    }

    void Push(int no)
    {
        PNODE newn =  new NODE;
        newn->data = no;
        newn->next = NULL;
        if(first==NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
        iCount++;
    }
    int Pop()
    {
        int ivalue;
          if(first==NULL)
        {
            cout<<"stack is empty"<<endl;
            return (int)0;
        }   
        else
        {
            PNODE temp = first;
            ivalue = temp->data;
            first = first->next;
            delete temp;  
            iCount--;
            return ivalue;
        }
    }
    void display()
    {
        if(first==NULL)
        {
            cout<<"stack is empty"<<endl;
            return;
        }
        PNODE temp = first;
        while(temp !=NULL)
        {
            cout<<"|"<<temp->data<<"|"<<endl;
            temp  = temp ->next;
        }
    }
    int count()
    {
        return iCount;
    }
};
int main()
{
    stackx ss;
    int ch = 0;
    int iNo = 0;
    int iRet = 0;
    int iPop = 0;
    do {
    cout<<"\n";
    cout<<"------------Menu---------------\n";
    cout<<"1.PUSH NUMBER\n";
    cout<<"2.POP NUMBER\n";
    cout<<"3.SHOW NO OF ELEMENTS\n";
    cout<<"4.DISPLAY\n";
    cout<<"5.EXIT\n";
    cout<<"Enter Your Choice:";
    cin>>ch;
    if(ch<1 || ch>5)
    {
        cout<<"INVALID INPUT";
    }
    switch (ch)
    {
       case 1:
       cout<<"Enter number you want to insert";
       cin>>iNo;
       ss.Push(iNo);
       break;
       case 2 :
       iPop=ss.Pop();
       cout<<"Element poped from stack:"<<iPop;
       break;
       case 3 :
       iRet = ss.count();
       cout<<"Number of elements :"<<iRet;
       break;
       case 4:
       ss.display();
       break;
       case 5:
       cout<<"Exisiting program....";
       break;
    }
    }while(ch!=5);
    return 0;
}

File: STACK/Stack.java
import java.util.Scanner;

class node {
    int data;
    node next;
}

class Stackx {
    public node first;
    public int icount;

    public Stackx() {
        first = null;
        icount = 0;
    }

    public void Push(int no) {
        node newn = new node();
        newn.data = no;
        newn.next = null;
        if (first == null) {
            first = newn;
        } else {
            newn.next = first;
            first = newn;
        }
        icount++;
    }

    public int Pop() // deletefirst
    {
        if (first == null) {
            System.out.println("stack is empty");
            return 0;
        } else {
            int ivalue = first.data;
            first = first.next;
            icount--;
            return ivalue;
        }
    }

    public void Display() {
        if (first == null) {
            System.out.println("stack is empty");
            return;
        }
        node temp = first;
        while (temp != null) {
            System.out.println("|" + temp.data + "|");
            temp = temp.next;
        }

    }

    public int Count() {
        return icount;
    }
}

class stack {
    public static void main(String arg[]) {
        Stackx sobj = new Stackx();
        int iRet = 0;
        int iNo = 0;
        int iCnt = 0;
        int ch = 0;
        Scanner s = new Scanner(System.in);
        do {
            System.out.println("------------MENU------------------");
            System.out.println("1.PUSH NUMBER");
            System.out.println("2.POP NUMBER");
            System.out.println("3.COUNT");
            System.out.println("4.DISPLAY");
            System.out.println("5.EXIT");
            System.out.println("PLEASE ENTER YOUR CHOICE ");
            ch = s.nextInt();
            if (ch < 1 || ch > 5) {
                System.out.println("invalid input");
            }
            System.out.println();
            switch (ch) {

                case 1:
                    System.out.println("Enter Number you want to PUSH:");
                    iNo = s.nextInt();
                    sobj.Push(iNo);
                    break;
                case 2:
                    iRet = sobj.Pop();
                    if (iRet > 0 || sobj.Count() > 0) {
                        System.out.println("poped elemnts is :" + iRet);
                    }
                    break;
                case 3:
                    iCnt = sobj.Count();
                    System.out.println("Number of elements in stack :" + iCnt);
                    break;
                case 4:
                    sobj.Display();
                    break;
                case 5:
                    System.out.println("Existing program");

            }

        } while (ch != 5);
    }
}
